src/dynamic_array.c

```diff
@@ -2,21 +2,19 @@
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
-DynamicArray initialize(size_t initial_capacity, size_t initial_size) {
-    if (initial_size > initial_capacity) {
-        fprintf(stderr,
-                "Initial size cannot be greater than initial capacity.\n");
-        exit(EXIT_FAILURE);
-    }
-    DynamicArray dy_arr;
+DynamicArray initialize(int *initial_values, size_t initial_size) {
+    size_t initial_capacity = GROWTH_FACTOR * initial_size;
```

Comment:
1) if initial_values is NULL, dont worry about initial_size, lets initialize empty array (for this you can create static function lilke "initEmpty" in .c file, because it is not about you API, it is private function of your array)
2) if initial_values is not NULL, check initial_size, if it zero, call initEmpty func :)

p.s. of cource if you call "initEmpty", you need to return instantly